parking/backend/engine/alloc_engine.py

```diff
@@ -0,0 +1,58 @@
+import logging
+from typing import Optional
+from geopy.distance import distance
+from parking.backend.db.dbaccess import DbAccess
+from parking.shared.rest_models import ParkingLot
+from parking.shared.location import Location
+from parking.shared.ws_models import ParkingRequestMessage
+
+logger = logging.getLogger('backend')
+
+
+class AllocationEngine():
+    def __init__(self, dba: DbAccess, user_sessions: 'UserSessions') -> None:
```

Comment:
Why is `UserSessions` in quotations here rather than just imported?

Reply:
There was a circular dependency between engine and wsserver (where UserSessions is).

@manfredlift has now split UserSessions into its own file, so this won't be an issue now.